### PR TITLE
V5.0 add timeseries: operations

### PR DESCRIPTION
**Pull request by IdanHaim**
TimeSeriesBatchOperation,
TimeSeriesOperation (Append and Remove),
GetTimeSeriesOperation,
Fix some small issues

### CODE REVIEW

File: pyravendb/raven_operations/timeseries_operations.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            self.url = (f"{server_node.url}/databases/{server_node.database}"
+                        f"/timeseries?id={Utils.quote_key(self._document_id)}"
+                        f"{f'&start={self._start}' if self._start > 0 else ''}"
+                        f"{f'&pageSize={self._page_size}' if self._page_size < sys.maxsize else ''}")
+
+            if self._ranges is not None:
+                for range_ in self._ranges:
+                    self.url += (f"&name={Utils.quote_key(range_.name)}"
+                                 f"&from={Utils.datetime_to_string(range_.from_date)}"
+                                 f"&to={Utils.datetime_to_string(range_.to_date)}")
```
Comment (ayende) [inline pyravendb/raven_operations/timeseries_operations.py]: If the user passed `None`, what do we get here?

File: pyravendb/raven_operations/timeseries_operations.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+import sys
+
+
+class TimeSeriesRange:
+    def __init__(self, name, from_date: Optional[datetime] = None, to_date: Optional[datetime] = None):
+        if not name:
+            raise ValueError("TimeSeriesRange must include name")
+        self.name = name
+        self.from_date = from_date
```
Comment (ayende) [inline pyravendb/raven_operations/timeseries_operations.py:15]: This should default to `0001-01-01`

File: pyravendb/raven_operations/timeseries_operations.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        self.to_date = to_date
```
Comment (ayende) [inline pyravendb/raven_operations/timeseries_operations.py:16]: This should default to `9999-12-31` when set to `None`

File: pyravendb/raven_operations/timeseries_operations.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+            if not document_id:
+                raise ValueError("Invalid document_id")
+
+            self._document_id = document_id
+            self._operation = operation
+
+        def create_request(self, server_node):
+            self.url = (f"{server_node.url}/databases/{server_node.database}"
+                        f"/timeseries?id={Utils.quote_key(self._document_id)}")
```
Comment (ayende) [inline pyravendb/raven_operations/timeseries_operations.py:160]: Does this work? I thought we need to send a `DocumentId` field in the JSON

**Review by ayende — CHANGES_REQUESTED** [review]
Can you also add proper tests for this functionality?

File: pyravendb/raven_operations/timeseries_operations.py (hunk 4 above)
Comment (IdanHaim) [reply pyravendb/raven_operations/timeseries_operations.py:160]: Yes I tested everything, this is exactly like the c# client, we send the document id in the URL

File: pyravendb/raven_operations/timeseries_operations.py (hunk 1 above)
Comment (IdanHaim) [reply pyravendb/raven_operations/timeseries_operations.py]: leftovers fixed, ranges are mandatory.

File: pyravendb/raven_operations/timeseries_operations.py (hunk 2 above)
Comment (IdanHaim) [reply pyravendb/raven_operations/timeseries_operations.py:15]: not needed if you don't send any will be automatically like this 
{'From': '0001-01-01T00:00:00.0000000Z', 'To': '9999-12-31T23:59:59.9999999Z'} example from the code

File: pyravendb/raven_operations/timeseries_operations.py (hunk 3 above)
Comment (IdanHaim) [reply pyravendb/raven_operations/timeseries_operations.py:16]: same

File: pyravendb/raven_operations/timeseries_operations.py (hunk 1 above)
Comment (IdanHaim) [reply pyravendb/raven_operations/timeseries_operations.py]: you can give only a name and it still works.
the server will return with all the time-series he has with the name

File: pyravendb/raven_operations/timeseries_operations.py (hunk 2 above)
Comment (ayende) [reply pyravendb/raven_operations/timeseries_operations.py:15]: How does this work? Both of them are: `Optional[datetime] = None` and are used with:

`from={Utils.datetime_to_string(range_.from_date)}`

Comment (ayende) [reply pyravendb/raven_operations/timeseries_operations.py:15]: Oh, wait, this happens on the server side, no?

**Pull request merged**